reporter/m3aggregator/reporter.go

```diff
@@ -119,16 +119,19 @@ func (r *reporter) ReportCounter(id id.ID, value int64) error {
 	)
 	r.incrementReportPending()
 	counter := unaggregated.Counter{ID: id.Bytes(), Value: value}
-	matchRes := r.matcher.ForwardMatch(id, fromNanos, toNanos)
-	if err := r.client.WriteUntimedCounter(
-		counter,
-		matchRes.ForExistingIDAt(fromNanos),
-	); err != nil {
-		multiErr = multiErr.Add(err)
+	matchResult := r.matcher.ForwardMatch(id, fromNanos, toNanos)
+
+	matchExisting := matchResult.ForExistingIDAt(fromNanos)
+	if !matchExisting.IsDropPolicyApplied() {
```

Comment:
hmm, not sure if I missed anything, say fromNanos is T0 and toNanos is T2, if there is a drop rule created at T1 (between T0 and T2), then the matched result should be [T0, T1+SomeStoragePolicy) and [T1, T2+Drop). So the collector should still be sending the metric for [T0, T1) instead of dropping the write? So the aggregator can look at what the correct time is to decide whether or not to take the write?